### PR TITLE
[FEATURE][TPCDS] Add white list help run the specified queries

### PR DESCRIPTION
**Pull request by zwangsheng**
### _Why are the changes needed?_

Add `white-list` help run the specified queries.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

  `--white-list q4,q15,q19,q25,q42,q49,q60,q66,q68,q72,q76,q78,q79,q89,q97`

popo_2022-09-22  15-01-05

### CODE REVIEW

**Comment by zwangsheng** [comment]
cc @pan3793 @yaooqinn

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala [hunk 1]
```diff
@@ -65,6 +66,9 @@ object RunBenchmark {
       opt[String]('r', "results-dir")
         .action((x, c) => c.copy(resultsDir = x))
         .text("dir to store benchmark results, e.g. hdfs://hdfs-nn:9870/pref")
+      opt[String]('w', "white-list")
```
Comment (pan3793) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala:69]: how about 'q' "queries" ? cc @ulysses-you

Comment (zwangsheng) [reply dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala:69]: Fine with `queries`, will upload later.

Comment (ulysses-you) [reply dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala:69]: > how about 'q' "queries" 

+1

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala [hunk 2]
```diff
@@ -65,6 +66,9 @@ object RunBenchmark {
       opt[String]('r', "results-dir")
         .action((x, c) => c.copy(resultsDir = x))
         .text("dir to store benchmark results, e.g. hdfs://hdfs-nn:9870/pref")
+      opt[String]('q', "queries")
+        .action((x, c) => c.copy(queries = Some(x)))
```
Comment (pan3793) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala:70]: let's parse and transform the parameter here, fail-fast if it's illegal

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala [hunk 3]
```diff
@@ -29,7 +29,8 @@ case class RunConfig(
     filter: Option[String] = None,
     iterations: Int = 3,
     breakdown: Boolean = false,
-    resultsDir: String = "/spark/sql/performance")
+    resultsDir: String = "/spark/sql/performance",
+    queries: Set[String])
```
Comment (pan3793) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala:33]:
```suggestion
    queries: Set[String] = Set.empty)
```

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala [hunk 4]
```diff
@@ -65,6 +66,16 @@ object RunBenchmark {
...
         .text("dir to store benchmark results, e.g. hdfs://hdfs-nn:9870/pref")
+      opt[String]('q', "queries")
+        .action((x, c) =>
+          c.copy(queries = {
+            if (x.nonEmpty || !x.isBlank) {
+              x.split(",").filter(_.nonEmpty).toSet
+            } else {
+              Set.empty[String]
+            }
+          }))
```
Comment (pan3793) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala:77]:
```suggestion
        .action { case (x, c) =>
          c.copy(queries = x.split(",").map(_.trim).filter(_.nonEmpty).toSet)
        }
```

File: dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala [hunk 5]
```diff
@@ -96,11 +107,13 @@ object RunBenchmark {
       benchmark.tpcds2_4Queries
     }
 
+    val runQueries = allQueries.filter(q => config.queries.contains(q.name.split('-')(0)))
```
Comment (pan3793) [inline dev/kyuubi-tpcds/src/main/scala/org/apache/kyuubi/tpcds/benchmark/RunBenchmark.scala:110]: we should retain the previous behavior if `config.queries` is empty.

**Comment by zwangsheng** [comment]
Unrelated CI failure.

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Thanks, merging to master/1.6

**Pull request closed without merging**